cmd/scrambled-eggs/scramble.py

```diff
@@ -7,20 +7,32 @@ def main():
     '''takes a json file as input and destructively hashes all string values in it'''
     args = argparse.ArgumentParser()
     args.add_argument("file")
+    args.add_argument("--multi", type=bool, nargs="?", const=True, default=False)
     flags = args.parse_args()
 
-    with open(flags.file, "r") as f:
-        data = json.load(f)
-
-    traverse(data)
-    
-    with open(flags.file, "w") as f:
-        json.dump(data, f)
+    if flags.multi:
+        processed = []
+        with open(flags.file, "r") as f:
+            for line in f:
+                line_data = line.strip()
+                data = json.loads(line_data)
+                traverse(data)
+                processed.append(data)
+        with open(flags.file, "w") as f:
+            for line in processed:
+                json.dump(line, f)
+                f.write("\n")
```

Comment:
For the record, you could also probably do something like:
```python
    if flags.multiline:
        with open(flags.file, "r") as f:
            for line in f:
                documents.append(json.loads(line))
    else:
        with open(flags.file, "r") as f:
            documents.append = json.load(f)

    for data in documents:
        traverse(data)

    with open(flags.file, "w") as f:
        for data in documents:
            json.dump(data, f, indent=2)
```
But I think this is less likely to introduce strange formatting changes.